src/zh/toptoon/src/eu/kanade/tachiyomi/extension/zh/toptoon/ToptoonDto.kt

```diff
@@ -0,0 +1,55 @@
+package eu.kanade.tachiyomi.extension.zh.toptoon
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.JsonDecoder
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.jsonPrimitive
+
+@Serializable
+data class PopularResponseDto(val adult: List<MangaDto>)
+
+@Serializable
+data class MangaDto(val meta: MetaDto, val thumbnail: ThumbnailDto, val id: String, val lastUpdated: LastUpdatedDto) {
+    val url = "/comic/epList/$id"
+}
+
+@Serializable
+data class MetaDto(val title: String, val author: AuthorDto)
+
+@Serializable
+data class AuthorDto(val authorString: String)
+
+@Serializable
+// "standard" in json can be either string or array, always decode as string
+data class ThumbnailDto(@Serializable(with = StringOrArraySerializer::class) val standard: String) {
+    val url = "https://tw-contents-image.toptoon.net$standard"
```

Comment:
1. Make the `standard` field private.
2. Should do `val url get() = ...`.
3. Use `val standard: JsonElement` and do the `when` in `val url get()`.